libfwupdplugin/fu-context.c

```diff
@@ -1001,6 +1001,33 @@ fu_context_housekeeping(FuContext *self)
 
 typedef gboolean (*FuContextHwidsSetupFunc)(FuContext *self, FuHwids *hwids, GError **error);
 
+static void
+fu_context_detect_full_disk_encryption(FuContext *self)
+{
+	FuContextPrivate *priv = GET_PRIVATE(self);
+	g_autoptr(GPtrArray) devices = NULL;
+	g_autoptr(GError) error_local = NULL;
+
+	g_return_if_fail(FU_IS_CONTEXT(self));
+
+	devices = fu_common_get_block_devices(&error_local);
+	if (devices == NULL) {
+		g_info("Failed to get block devices: %s", error_local->message);
+		return;
+	}
```

Comment:
I think this warrants a separate state flag indicating that FDE presence could not be determined. Otherwise it's impossible to distinguish lack of FDE from a case where it could be present but udisks isn't available.

```
typedef enum {
	/**
	 * No FDE was detected.
	 */
	FU_CONTEXT_FLAG_NO_FDE = 0, /** no FDE detected */
	 /**
	 *  FDE presence could not be checked for due to other errors.
	 */
	FU_CONTEXT_FLAG_UNDEFINED_FDE = 1 << 1,
	/**
	  * Found volumes using BitLocker.
	  */
	FU_CONTEXT_FLAG_BITLOCKER_FDE, = 1 << 2
} f_context_fde_t;


guint fu_context_get_full_disk_encryption(FuContext *self) {
	priv->fde_flag;
}
```

then here
```suggestion
	if (devices == NULL) {
		g_info("Failed to get block devices: %s", error_local->message);
		/* there could be volumes using FDE, but we were unable to enumerate
		 * devices, so stay on the safe side and let other code decide what to do */
		priv->fde_flag = FU_CONTEXT_FLAG_UNDEFINED_FDE;
		return;
	}
```

Reply:
I personally think if the user is rolling their own distro without udisks then it's on their head to work out integration issues like this.